Sources/Benchmark/BenchmarkColumn.swift

```diff
@@ -0,0 +1,242 @@
+            columns.append(
+                BenchmarkColumn(
+                    name: counter,
+                    content: { result in
+                        if let value = result.counters[counter] {
+                            return .number(value)
+                        } else {
+                            return .number(0)
+                        }
+                    },
+                    alignment: .right))
+        }
+
+        return columns
+    }
+
+    static func evaluate(columns: [Column], results: [BenchmarkResult], pretty: Bool) -> [Row] {
+        var header: Row = [:]
+        for column in columns {
+            header[column] = column.name
+        }
+
+        var rows: [Row] = [header]
+        for result in results {
+            var row: Row = [:]
+            for column in columns {
+                var content: String
+                if !pretty {
+                    switch column.content(result) {
+                    case .string(let value):
+                        content = value
+                    case .time(let value):
+                        content = String(value)
+                    case .inverseTime(let value):
+                        content = String(value)
+                    case .percentage(let value):
+                        content = String(value)
+                    case .number(let value):
+                        content = String(value)
+                    }
+                } else {
+                    switch column.content(result) {
+                    case .string(let value):
+                        content = value
+                    case .time(let value):
+                        switch result.settings.timeUnit {
+                        case .ns: content = "\(value) ns"
+                        case .us: content = "\(value/1000.0) us"
+                        case .ms: content = "\(value/1000_000.0) ms"
+                        case .s: content = "\(value/1000_000_000.0) s"
+                        }
+                    case .inverseTime(let value):
+                        switch result.settings.inverseTimeUnit {
+                        case .ns: content = "\(value) /ns"
+                        case .us: content = "\(value*1000.0) /us"
+                        case .ms: content = "\(value*1000_000.0) /ms"
+                        case .s: content = "\(value*1000_000_000.0) /s"
+                        }
+                    case .percentage(let value):
+                        content = String(format: "%6.2f %%", value)
+                    case .number(let value):
+                        let string = String(value)
+                        if string.hasSuffix(".0") {
+                            content = String(string.dropLast(2))
+                        } else {
+                            content = string
+                        }
+                    }
+                    if column.name == "std" || column.name == "std_abs" {
```

Comment:
I'm okay with this being hard-coded for now, but it might be good to generalize in the future if we discover a good abstraction here.

Reply:
Generalized in 0c30506 via formatter functions.